api/config/v1alpha1/envoygateway_types.go

```diff
@@ -42,6 +42,11 @@ type EnvoyGatewaySpec struct {
 	// +optional
 	Provider *EnvoyGatewayProvider `json:"provider,omitempty"`
 
+	// XDSServer defines the desired xds server configuration.
+	//
+	// +optional
+	XDSServer *XDSServer `json:"xdsServer,omitempty"`
```

Comment:
prefer `ControlPlaneAuth *ControlPlaneTLSConfig` over `xdsServer` because the ca cert needs to be programed in xds client / envoy proxy, but hoping others chime in here

Reply:
looks like this field needs to exist in `EnvoyGateway` and `EnvoyProxy` since we have certs/keys for server and client
https://github.com/envoyproxy/gateway/blob/0f710c430e29c741d376f799c101a2f11a058fc4/internal/crypto/certgen.go#L141

Reply:
Open to any naming suggestions 😄. But I do want to clarify that the `XDSServer` is for configuring the server side tls config not the client side. The client side would currently be configured through the EnvoyProxy CR under `EnvoyProxyKubernetesProvider.TLSConfigSecret`.

Reply:
thinking about this again, rather than adding it in two places, does it make sense to just add it here with two different top level fields, keeping all the control plane auth config in one place ?

Reply:
thoughts on this style
```
controlPlaneAuth:
  envoyGatewayCertRef:
  envoyProxyCertRef:
  caCertRef:
```